xc/device.ml

```diff
@@ -1502,38 +1531,18 @@ module Vusb = struct
       (fun () ->
          let is_running = Qemu.is_running ~xs domid in
          if is_running then
-           qmp_send_cmd domid Qmp.(Device_del id) |> ignore
+           try
+             qmp_send_cmd domid Qmp.(Device_del id) |> ignore
+           with QMP_connection_error ->
+             raise Device_not_connected
       )
       (fun () ->
-         usb_reset_detach ~hostbus ~hostport ~domid ~privileged
+         let peripherals = qom_list ~xs ~domid in
+         let found = List.mem id peripherals in
+         if found then
```

Comment:
I would keep it more compact. I don't think the extra names add much value to clarity and they are not reused.

```
if List.mem id (qom_list ~xs ~domid) then
```

Reply:
Ok. Make sense